tests/visualization_tests/matplotlib_tests/test_edf.py

```diff
@@ -76,3 +78,59 @@ def test_plot_optimization_history(direction: str) -> None:
     _validate_edf_values(lines[0].get_ydata())
     assert len(figure.get_lines()) == 1
     assert figure.xaxis.label.get_text() == "Target Name"
+
+
+@pytest.mark.parametrize("value", [float("inf"), -float("inf"), float("nan")])
+def test_nonfinite_removed(value: int) -> None:
+
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"x": 1.0},
+            distributions={"x": FloatDistribution(0.0, 1.0)},
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=value,
+            params={"x": 0.0},
+            distributions={"x": FloatDistribution(0.0, 1.0)},
+        )
+    )
+
+    figure = plot_edf(study)
+    assert all(np.isfinite(figure.get_lines()[0].get_xdata()))
```

Comment:
This assertion was fine on the latest master branch as well, we might need to finer assertion. I tried `figure.findobj(PathCollection)` by following `test_slice.py`, but it returned empty.

Note that Plotly's tests look perfect.

Reply:
I think this might be a difficult issue; we can work on this as a follow-up and keep the tests as is in this PR.

Reply:
This was my mistake (actually I checked on this branch, not on the master). The test was perfect. Sorry for confusing you.

Reply:
Sounds good to me!